File: LeetCode-Programs/3Sum Closest.cpp
class Solution {
public:
    int threeSumClosest(vector<int>& nums, int target) {
        
       sort(nums.begin(),nums.end());
        int res=99999999,sum=0;
        for(int i=0;i<nums.size()-2;i++)
        {
            int j=i+1;
            int k=nums.size()-1;
            while(j<k)
            {
                sum=nums[i]+nums[j]+nums[k];
                if (abs(target-sum) < abs(target-res)) 
                {
				         res=sum;
                }
                if(sum>target)
                {
                    k--;
                }
                else
                {
                    j++;
                }
             }
        }
            return res;
    }
};

File: LeetCode-Programs/3Sum Closest.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        nums.sort()
        res = nums[0] + nums[1] + nums[2]
        for i in range(len(nums)):
            j=i+1 
            k=len(nums) - 1
            while j<k:
                sum=nums[i]+nums[j]+nums[k]
                if abs(target-res)>abs(target-sum): 
                    res=sum
                if sum<target: 
                    j+=1
                elif sum>target: 
                    k-=1
                elif sum==target: 
                    return target
        return res
